src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -1,23 +1,39 @@
 import {createStackNavigator} from '@react-navigation/stack';
 import React from 'react';
 import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
 import NoDropZone from '@components/DragAndDrop/NoDropZone';
 import useOnboardingLayout from '@hooks/useOnboardingLayout';
 import useThemeStyles from '@hooks/useThemeStyles';
 import OnboardingModalNavigatorScreenOptions from '@libs/Navigation/AppNavigator/OnboardingModalNavigatorScreenOptions';
+import Navigation from '@libs/Navigation/Navigation';
 import type {OnboardingModalNavigatorParamList} from '@libs/Navigation/types';
 import OnboardingPersonalDetails from '@pages/OnboardingPersonalDetails';
 import OnboardingPurpose from '@pages/OnboardingPurpose';
 import OnboardingWork from '@pages/OnboardingWork';
+import * as Report from '@userActions/Report';
+import ONYXKEYS from '@src/ONYXKEYS';
 import SCREENS from '@src/SCREENS';
 import Overlay from './Overlay';
 
+type OnboardingModalNavigatorProps = {
+    /** Current onboarding completion status */
+    hasCompletedGuidedSetupFlow: boolean;
+};
+
 const Stack = createStackNavigator<OnboardingModalNavigatorParamList>();
 
-function OnboardingModalNavigator() {
+function OnboardingModalNavigator({hasCompletedGuidedSetupFlow}: OnboardingModalNavigatorProps) {
     const styles = useThemeStyles();
     const {shouldUseNarrowLayout} = useOnboardingLayout();
 
+    if (hasCompletedGuidedSetupFlow) {
+        Navigation.goBack();
+        Report.navigateToConciergeChat();
+        // eslint-disable-next-line react/jsx-no-useless-fragment
+        return <></>;
```

Comment:
```suggestion
        return null;
```

Reply:
For me, when I test it on IOS and android, `return null` throws some errors. That's why I put Fragment. Need to test again though.